Automattic/pocket-casts-ios
# Tracks: Add player shelf events

| 📘 Project: #154 | 🛫 Depends on: #292 |
|:---:|:---:|

This adds events for the "player shelf" actions. The shelf is the view that appears at the bottom of the player and contains the customizable player action buttons. The word shelf is used in the codebase by both Android and iOS.

- `player_shelf_action_tapped`: When the user taps on one of the actions located in the player shelf or overflow menu
- `player_shelf_overflow_menu_shown`: When the user taps on the `...` button and the overflow menu is displayed
- `player_shelf_overflow_menu_rearrange_started`: When the user taps the Edit button and the overflow menu enters rearrange mode
- `player_shelf_overflow_menu_rearrange_action_moved`: When the user moves an action in during a rearrange
- `player_shelf_overflow_menu_rearrange_finished`: When the user exits the rearrange mode

This also adds some generic events that will be used elsewhere in the app: 

- `episode_starred`: When an episode is starred
- `episode_archived`: When an episode is archived
- `podcast_shared`: When a podcast is shared

## To test

1. Enable the `tracksEnabled` option in FeatureFlag.swift
2. Play an episode if you're not already
3. Open the full screen player
4. Tap each of the 4 actions located in the shelf at the bottom of the player view
5. ✅ Verify you see the following in console: `🔵 Tracked: player_shelf_action_tapped ["from": "shelf", "action": "NAME"]` - Where name is the name of the action that was tapped
6. Tap the ... to open the overflow menu
7. ✅ `🔵 Tracked: player_shelf_overflow_menu_shown`
8. Tap the actions located in the overflow menu
9. ✅ `🔵 Tracked: player_shelf_action_tapped ["from": "overflow_menu", "action": "NAME"]`
10. Open the overflow menu again
11. Tap the edit button
12. ✅ `🔵 Tracked: player_shelf_overflow_menu_rearrange_started`
13. Move an item out of the "Shortcut on player" section
14. ✅ `🔵 Tracked: player_shelf_overflow_menu_rearrange_action_moved ["action": "NAME", "moved_to": "overflow_menu", "moved_from": "shelf", "position": POSITION]`
    - NAME: Is the name of the action that was moved
    - POSITION is the new position in section it was moved to
15. Move an action from the "IN MENU" section to the "Shortcut on Player" section
16. ✅ `🔵 Tracked: player_shelf_overflow_menu_rearrange_action_moved ["moved_from": "overflow_menu", "action": "NAME", "moved_to": "shelf", "position": POSITION]`
17. Move an item to a higher position in the Shortcut On Player section
18. ✅ `🔵 Tracked: player_shelf_overflow_menu_rearrange_action_moved ["moved_from": "shelf", "action": "NAME", "moved_to": "shelf", "position": POSITION]`
19. Tap the Done button
20. ✅ `🔵 Tracked: player_shelf_overflow_menu_rearrange_finished`

#### Episode starred / archived / podcast shared
1. Following the steps from above
2. Locate the Star action in the shelf, and tap it
4. ✅ `🔵 Tracked: episode_starred ["source": "player"]`
5. Tap the Share button
8. Select the Podcast item
9. ✅ `🔵 Tracked: podcast_shared ["type": "podcast", "source": "player"]`
10. Tap the Share button again, and tap the Episode item
11. ✅ `🔵 Tracked: podcast_shared ["type": "episode", "source": "player"]`
12. Tap the Share button again, and tap the Current position item
13. ✅ `🔵 Tracked: podcast_shared ["source": "player", "type": "current_position"]`
6. Tap Archive button, and confirm the archive
7. ✅ `🔵 Tracked: episode_archived ["source": "player"]`

## Checklist
